File: app/src/main/java/com/example/myappcontacts/business/contactsmap/ContactsMapInteractor.java
package com.example.myappcontacts.business.contactsmap;

import com.example.myappcontacts.data.dao.contactsmap.MapMarkersModel;
import com.example.myappcontacts.data.repositories.contactsmap.IContactsMapRepository;

import java.util.List;

import io.reactivex.Single;

public class ContactsMapInteractor implements IContactsMapInteractor {

    private IContactsMapRepository mIContactsMapRepository;

    public ContactsMapInteractor(IContactsMapRepository IContactsMapRepository) {
        mIContactsMapRepository = IContactsMapRepository;
    }

    @Override
    public Single<List<MapMarkersModel>> loadMapMarkersList() {
        return mIContactsMapRepository.loadMapMarkersList();
    }
}

File: app/src/main/java/com/example/myappcontacts/business/contactsmap/ContactsMapInteractor.kt
package com.example.myappcontacts.business.contactsmap

import com.example.myappcontacts.data.dao.contactsmap.db.MapMarkersModel
import com.example.myappcontacts.data.repositories.contactsmap.IContactsMapRepository

import io.reactivex.Single
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers

class ContactsMapInteractor(private val contactsMapRepository: IContactsMapRepository) : IContactsMapInteractor {

    override fun loadMapMarkersList(): Single<List<MapMarkersModel>> {
        return contactsMapRepository
                .loadMapMarkersList()
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
    }
}
